packages/app/obojobo-repository/shared/components/notification.test.js

```diff
@@ -0,0 +1,151 @@
+import React from 'react'
+import { create, act } from 'react-test-renderer'
+import Notification from './notification'
+
+describe('Notification component', () => {
+	beforeAll(() => {
+		Object.defineProperty(document, 'cookie', {
+			value: '',
+			writable: true
+		})
+	})
+
+	test('renders without crashing', () => {
+		const component = create(<Notification onDataFromNotification={() => {}} />) // Provide a mock function for onDataFromNotification
+		const tree = component.toJSON()
+		expect(tree).toMatchSnapshot()
+	})
+
+	test('loads notifications from cookies on mount', () => {
+		const onDataFromNotification = jest.fn()
+		const notificationValue = [{ key: 1, text: 'Test Notification', title: 'Test Title' }]
+		document.cookie = `notifications=${JSON.stringify(notificationValue)}`
+
+		const component = create(<Notification onDataFromNotification={onDataFromNotification} />)
+		const tree = component.toJSON()
+
+		expect(tree).toMatchSnapshot()
+		expect(document.cookie).toBe(
+			`notifications=${encodeURIComponent(JSON.stringify(notificationValue))};`
+		)
+	})
+
+	test('handles click on exit button and updates state and cookie', () => {
+		const onDataFromNotification = jest.fn()
+		const notificationValue = [
+			{ key: 1, text: 'Notification1', title: 'Title1' },
+			{ key: 2, text: 'Notification2', title: 'Title2' }
+		]
+		document.cookie = `notifications=${JSON.stringify(notificationValue)}`
+
+		const reusableComponent = <Notification onDataFromNotification={onDataFromNotification} />
+		let component
+		act(() => {
+			component = create(reusableComponent)
+		})
+
+		let tree = component.toJSON()
+		expect(tree).toMatchSnapshot()
+
+		// Simulate click on exit button for the first notification
+		const key = 0
+		const exitButtons = component.root.findAllByProps({ className: 'notification-exit-button' })
+
+		if (exitButtons[key]) {
```

Comment:
Sorry if I didn't comment on or notice this before. Feels like I keep sending you back for different things every time.

Is there any scenario where this test should run successfully and `exitButtons[key]` would evaluate falsy, or if `exitButtons[key]` is falsy does that mean something went wrong?

Based on the `findAllByProps` line above I would expect a case where `exitButtons[key]` not resolving to something truthy to indicate that there was a problem. In a case like this you could use `https://jestjs.io/docs/expect#tobetruthy`, or forego the `key` variable altogether and `expect(exitButtons.length).toBe(0)` or even `expect(exitButtons.length > 0).toBe(true)`.